From 215241962183399024fe323d2ba74b19d0b6b7f7 Mon Sep 17 00:00:00 2001
From: Zhen Chen <chenzhen@rock-chips.com>
Date: Tue, 16 Jan 2018 12:03:03 +0800
Subject: [PATCH] MALI: midgard: RK: rename output mali_kbase.ko to
 midgard_kbase.ko

Some optimizations on files of KBuild in addition.

Change-Id: I1db012e116b8b69897a2791ae610da35365a1a61
Signed-off-by: Zhen Chen <chenzhen@rock-chips.com>
---
 drivers/gpu/arm/midgard/Kbuild             | 18 +++++++-----------
 drivers/gpu/arm/midgard/platform/Kbuild    | 21 ---------------------
 drivers/gpu/arm/midgard/platform/rk/Kbuild | 12 ++----------
 3 files changed, 9 insertions(+), 42 deletions(-)
 delete mode 100644 drivers/gpu/arm/midgard/platform/Kbuild

diff --git a/drivers/gpu/arm/midgard/Kbuild b/drivers/gpu/arm/midgard/Kbuild
index 4348704e7640..34725d60995e 100644
--- a/drivers/gpu/arm/midgard/Kbuild
+++ b/drivers/gpu/arm/midgard/Kbuild
@@ -13,7 +13,6 @@
 #
 
 
-
 # Driver version string which is returned to userspace via an ioctl
 MALI_RELEASE_NAME ?= "r14p0-01rel0"
 
@@ -189,20 +188,17 @@ ifeq ($(CONFIG_MALI_PLATFORM_THIRDPARTY),y)
 platform_name := $(shell echo $(CONFIG_MALI_PLATFORM_THIRDPARTY_NAME))
 MALI_PLATFORM_THIRDPARTY_DIR := platform/$(platform_name)
 ccflags-y += -I$(src)/$(MALI_PLATFORM_THIRDPARTY_DIR)
-ifeq ($(CONFIG_MALI_MIDGARD),m)
+MALI_PLATFORM_DIR := platform/$(platform_name)
 include  $(src)/platform/$(platform_name)/Kbuild
-else ifeq ($(CONFIG_MALI_MIDGARD),y)
-obj-$(CONFIG_MALI_MIDGARD) += platform/
-endif
 endif
 
 # Tell the Linux build system from which .o file to create the kernel module
-obj-$(CONFIG_MALI_MIDGARD) += mali_kbase.o
+obj-$(CONFIG_MALI_MIDGARD) += midgard_kbase.o
 
 # Tell the Linux build system to enable building of our .c files
-mali_kbase-y := $(SRC:.c=.o)
+midgard_kbase-y := $(SRC:.c=.o)
 
-mali_kbase-$(CONFIG_MALI_DMA_FENCE) += mali_kbase_dma_fence.o
+midgard_kbase-$(CONFIG_MALI_DMA_FENCE) += mali_kbase_dma_fence.o
 
 MALI_BACKEND_PATH ?= backend
 CONFIG_MALI_BACKEND ?= gpu
@@ -211,12 +207,12 @@ CONFIG_MALI_BACKEND_REAL ?= $(CONFIG_MALI_BACKEND)
 ifeq ($(MALI_MOCK_TEST),1)
 ifeq ($(CONFIG_MALI_BACKEND_REAL),gpu)
 # Test functionality
-mali_kbase-y += tests/internal/src/mock/mali_kbase_pm_driver_mock.o
+midgard_kbase-y += tests/internal/src/mock/mali_kbase_pm_driver_mock.o
 endif
 endif
 
 include  $(src)/$(MALI_BACKEND_PATH)/$(CONFIG_MALI_BACKEND_REAL)/Kbuild
-mali_kbase-y += $(BACKEND:.c=.o)
+midgard_kbase-y += $(BACKEND:.c=.o)
 
 ccflags-y += -I$(src)/$(MALI_BACKEND_PATH)/$(CONFIG_MALI_BACKEND_REAL)
 subdir-ccflags-y += -I$(src)/$(MALI_BACKEND_PATH)/$(CONFIG_MALI_BACKEND_REAL)
@@ -227,7 +223,7 @@ ifeq ($(CONFIG_MALI_PLATFORM_THIRDPARTY)$(CONFIG_MALI_PLATFORM_FAKE),)
 CONFIG_MALI_PLATFORM_DEVICETREE := y
 endif
 
-mali_kbase-$(CONFIG_MALI_PLATFORM_DEVICETREE) += \
+midgard_kbase-$(CONFIG_MALI_PLATFORM_DEVICETREE) += \
 	platform/devicetree/mali_kbase_runtime_pm.o \
 	platform/devicetree/mali_kbase_config_devicetree.o
 ccflags-$(CONFIG_MALI_PLATFORM_DEVICETREE) += -I$(src)/platform/devicetree
diff --git a/drivers/gpu/arm/midgard/platform/Kbuild b/drivers/gpu/arm/midgard/platform/Kbuild
deleted file mode 100644
index 558657bbced9..000000000000
--- a/drivers/gpu/arm/midgard/platform/Kbuild
+++ /dev/null
@@ -1,21 +0,0 @@
-#
-# (C) COPYRIGHT 2012 ARM Limited. All rights reserved.
-#
-# This program is free software and is provided to you under the terms of the
-# GNU General Public License version 2 as published by the Free Software
-# Foundation, and any use by you of this program is subject to the terms
-# of such GNU licence.
-#
-# A copy of the licence is included with the program, and can also be obtained
-# from Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
-# Boston, MA  02110-1301, USA.
-#
-#
-
-
-
-ifeq ($(CONFIG_MALI_PLATFORM_THIRDPARTY),y)
-# remove begin and end quotes from the Kconfig string type
-	platform_name := $(shell echo $(CONFIG_MALI_PLATFORM_THIRDPARTY_NAME))
-	obj-y += $(platform_name)/
-endif
diff --git a/drivers/gpu/arm/midgard/platform/rk/Kbuild b/drivers/gpu/arm/midgard/platform/rk/Kbuild
index 700fd4341da5..95ca23bd9695 100755
--- a/drivers/gpu/arm/midgard/platform/rk/Kbuild
+++ b/drivers/gpu/arm/midgard/platform/rk/Kbuild
@@ -12,14 +12,6 @@
 #
 #
 
-ccflags-y += -I$(srctree)/drivers/staging/android
+SRC += \
+	$(MALI_PLATFORM_DIR)/mali_kbase_config_rk.o \
 
-ifeq ($(CONFIG_MALI_MIDGARD),y)
-obj-y += mali_kbase_config_rk.o
-# obj-y += mali_kbase_dvfs.o
-# obj-y += mali_kbase_platform.o
-else ifeq ($(CONFIG_MALI_MIDGARD),m)
-SRC += platform/rk/mali_kbase_config_rk.c
-# SRC += platform/rk/mali_kbase_dvfs.c
-# SRC += platform/rk/mali_kbase_platform.c
-endif
-- 
2.35.3

